George7h/investgrow-hub
# Merge into main

This React capstone project is about building a mobile web application to check a list of metrics (numeric values) that you will create making use of React and Redux.

You will select an API that provides numeric data about a topic that you like and then build the webapp around it. The webapp will have several pages:

one page with a list of items that could be filtered by some parameters; in the example below, it's a list of metrics that can be filtered by the country (imagine a search field to introduce the country name like Italy, Croatia, etc.). This page should be your homepage.
one page for the item details; in the example, the detail page for Czech Republic cities with number of views.

API First you need to choose an API to base the development of the webapp on. The API should allow you to get numeric values based on a parameter. We recommend that you choose an API that is open or needs an API key. If you choose an API that require authentication, you should implement it on your own.

Design

- You should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
- Colors (select one main color for your website).
- Typography: font face, size and weight.
- Layout: composition and space between elements.
- Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

Interactions

Home page

- When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
- Along with the category name, you will display some numeric values per category that come from the API.
- When the user clicks (or taps) on a category item, the application navigates to the details page.

Details page

In the details page, the webapp retrieves data from the API to show detailed data in the category.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

Testing requirements

- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
- You may need to mock the access to the API, so that your tests don't send actual requests.
- You may need to mock the connection to the Redux Store.


Technical requirements:

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- You should filter the data that you retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project should be deployed and accessible online using [Render](https://render.com/), [Netlify](https://www.netlify.com/) or [GitHub Pages](https://pages.github.com/).
- Add a link to the online version of your app to your README file.
